core/src/main/java/hudson/model/User.java

```diff
@@ -331,10 +347,24 @@ public synchronized void addProperty(@Nonnull UserProperty p) throws IOException
         } catch (DataAccessException e) {
             // seems like it's in the same boat as UserMayOrMayNotExistException
         }
+        
+        return null;
+    }
 
+    /**
+     * Creates an {@link Authentication} object that represents this user using the given userDetails
+     *
+     * @param userDetails Provided by {@link #getUserDetailsForImpersonation()}
+     * @see #getUserDetailsForImpersonation()
+     */
+    @Restricted(NoExternalUse.class)
+    public @Nonnull Authentication impersonate(UserDetails userDetails) {
```

Comment:
Could be the user details null?

Reply:
anonymous only, I suppose

Reply:
Taking a look at SecurityListener seems like this never should happen if the user is authenticated it should have details

Reply:
Actually the change in that code is simply a split of the existing method in order to retrieve the intermediate value.
The userDetails could be null in the case LDAP plugin is misconfigured or LDAP server is not reachable, the UserMayOrMayNotException is thrown and so in this case the returned userDetails  is null.